climpred/utils.py

```diff
@@ -68,3 +69,25 @@ def intersect(lst1, lst2):
     """
     lst3 = [value for value in lst1 if value in lst2]
     return np.array(lst3)
+
+
+def reduce_time_series(forecast, reference, nlags):
```

Comment:
We might need to also call `intersect` in here. Currently, it just reduces based on maximum and minimum lead, but does not check inits within that bound. What if someone has a subset of inits, or only initializes every N years?

Maybe this is dealt with implicitly in the list comprehension in `compute_hindcast`. Just a note for me to check on this.